### PR TITLE
feat: add un/archive dataset and reuse button

### PR DESCRIPTION
**Pull request by bolinocroustibat**
Closes https://github.com/datagouv/data.gouv.fr/issues/1393

EDIT: also added a "archived" badge UI on reuse card.

### CODE REVIEW

**Review by magopian — APPROVED** [review]
LGTM, nicely done 👍 

Do we need/want tests for that? I couldn't run tests locally, they fail with various phantomjs errors, i'll have to check that later.

**Comment by magopian** [comment]
Quick heads up, I believe there's a mistake in the PR description, #3104 is the PR number, not the original issue number?

File: js/views/dataset.vue [hunk 1]
```diff
@@ -169,6 +182,22 @@ export default {
         edit() {
             this.$go({name: 'dataset-edit', params: {oid: this.dataset.id}});
         },
+        archive() {
+        this.dataset.archived = new Date().toISOString();
```
Comment (maudetes) [inline js/views/dataset.vue:186]: Somehow the `archive` quickly shows but then disappears in the second (probably due to a watch effect)?

**Review by maudetes — COMMENTED** [review]
Thank you! :pray:

Could update the changelog entry made for reuse with this PR as well?

File: js/views/dataset.vue (hunk 1 above)
Comment (bolinocroustibat) [reply js/views/dataset.vue:186]: Couldn't reproduce... I lack a bit experience on this project JS part, any idea on why and how to fix that?

**Comment by bolinocroustibat** [comment]
> for that? I couldn't run tests locally, they fail with various phantomjs errors, i'll have to check that later.

It seems there is no tests for the JS part, aren't there?
Also I think the team is working on a complete frontend admin refactoring right now, so maybe it's not the right time to start to add tests to the front admin?

File: js/views/dataset.vue (hunk 1 above)
Comment (bolinocroustibat) [reply js/views/dataset.vue]: I've been debugging this issue but couldn't figure it out yet for now.
Notes:
- it seems the watcher is successfully triggered on any update on dataset.archived
- it seems the function `addOrRemoveBadge`works as expected as it add the "archived" badge to the badges list and removes it as expected
Could it be some display logic error elsewhere in a badges component?

Comment (bolinocroustibat) [reply js/views/dataset.vue]: BTW logic to add a "archived" badge on reuse page as well as been added to the PR.

Comment (maudetes) [reply js/views/dataset.vue]: Finally understood the issue! It came from the return value beeing not formatted exactly as ISO and thus triggering the watch effect again. Fixed it by checking if the value is None or not when removing the badge : https://github.com/opendatateam/udata/pull/3104/commits/6107ab52c9f13318597818db6c3d75563f923b1c.

Comment (bolinocroustibat) [reply js/views/dataset.vue]: @maudetes Amazing, well done! Shall I merge from main and merge?

**Review by maudetes — APPROVED** [review]
I think we're good now! Thank you :pray:

**Pull request merged**